filters/include/pcl/filters/impl/uniform_sampling.hpp

```diff
@@ -43,19 +43,13 @@
 
 //////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 template <typename PointT> void
-pcl::UniformSampling<PointT>::applyFilter (PointCloud &output)
+pcl::UniformSampling<PointT>::applyFilter (Indices &indices)
 {
-  // Has the input dataset been set already?
-  if (!input_)
-  {
-    PCL_WARN ("[pcl::%s::applyFilter] No input dataset given!\n", getClassName ().c_str ());
-    output.width = output.height = 0;
-    output.clear ();
-    return;
-  }
+  // The arrays to be used
+  indices.resize (indices_->size ());
```

Comment:
I think `indices` could be instead resized to `leaves_.size()` just before the second pass. That way, we have a better, "tighter" estimate of the final size.

Reply:
leaves_ is a std::unordered_map without init size. This also results in the order of the indexes calculated each time being uncertain.

Reply:
Sorry, I might have been unclear. I meant in line 132, just before iterating over `leaves_` in the for-loop. As far as I can see, there is nothing inserted into `indices` before that for-loop.
The order of the indices is random, not matter where `indices` is resized, right?

Reply:
Updated